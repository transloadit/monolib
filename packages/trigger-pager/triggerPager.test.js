const { api } = require('@pagerduty/pdjs')
const triggerPager = require('./triggerPager')

jest.mock('@pagerduty/pdjs')

describe('triggerPager', () => {
  test('main', async () => {
    const post = jest.fn(async (endpoint, payload) => {
      expect(endpoint).toBe('/incidents')
      expect(payload).toMatchSnapshot()
      return { data: { error: null } }
    })
    api.mockReturnValue({ post })
    await triggerPager({
      title      : 'us-east-1 down',
      description: 'this happens every 2 years',
      serviceId  : 'SERVICE_ID',
    })
    expect(post.mock.calls.length).toBe(1)
  })

  test('error', async () => {
    const post = jest.fn(async () => {
      return {
        data: {
          error: {
            message: 'oh no',
            errors : ['oh', 'no'],
          },
        },
      }
    })
    api.mockReturnValue({ post })

    let err
    try {
      await triggerPager({
        title      : '',
        description: '',
      })
    } catch (_err) {
      err = _err
    }

    expect(err.message).toBe('oh no - oh; no')
  })

  test('duplicate incident', async () => {
    const post = jest.fn(async () => {
      return {
        data: {
          error: {
            message: 'matching dedup key already exists',
            errors : [],
          },
        },
      }
    })
    api.mockReturnValue({ post })

    await triggerPager({
      title      : '',
      description: '',
    })
  })
})
