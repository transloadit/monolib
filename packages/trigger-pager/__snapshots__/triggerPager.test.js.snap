// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`triggerPager main 1`] = `
Object {
  "data": Object {
    "incident": Object {
      "body": Object {
        "details": "this happens every 2 years",
        "type": "incident_body",
      },
      "incident_key": undefined,
      "priority": Object {
        "id": "PUTY3A1",
        "type": "priority_reference",
      },
      "service": Object {
        "id": "serviceId",
        "type": "service_reference",
      },
      "title": "us-east-1 down",
      "type": "incident",
      "urgency": "high",
    },
  },
  "headers": Object {
    "from": "billing@transloadit.com",
  },
}
`;
